src/liballoc/collections/btree/map.rs

```diff
@@ -1716,16 +1780,15 @@ impl<'a, K, V> Iterator for RangeMut<'a, K, V> {
 }
 
 impl<'a, K, V> RangeMut<'a, K, V> {
-    unsafe fn next_unchecked(&mut self) -> (&'a K, &'a mut V) {
+    unsafe fn next_unchecked(&mut self) -> (&'a mut K, &'a mut V) {
```

Comment:
It's fine since we are replacing all the keys here and the resulting order is consistent.

Reply:
Should we document on `next_unchecked` that that's something callers need to ensure if they're going to mutate keys?

Reply:
I'm not sure it's necessary? It was already possible to mutate keys via `into_kv_mut` within the mod, this is just a private convenient iterator over those mutable references. It's not immediately clear that `next_unchecked` on a `RangeMut` would let you violate the `BTree` invariants where `next` would not, though, so I'm happy to add something documenting that for future internal use! It might be good to also document `into_kv_mut` with respect to that, in that case, because it has the same risk and is used frequently.